src/components/reusable/datePicker/datepicker.stories.js

```diff
@@ -1,114 +1,116 @@
-import { html } from 'lit';
 import './index';
+import { html } from 'lit';
 import { action } from '@storybook/addon-actions';
-import { ifDefined } from 'lit/directives/if-defined.js';
-import '@kyndryl-design-system/shidoka-foundation/components/icon';
-import { createOptionsArray } from '../../../common/helpers/helpers';
-import { DATE_PICKER_TYPES } from './defs';
-
-const createSelectOptions = (defs) => [null, ...createOptionsArray(defs)];
+import { useEffect } from '@storybook/addons';
 
 export default {
   title: 'Components/DatePicker',
   component: 'kyn-date-picker',
   parameters: {
     design: {
       type: 'figma',
-      url: 'https://www.figma.com/file/6AovH7Iay9Y7BkpoL5975s/Component-Library-for-Dev?node-id=508%3A142381&mode=dev',
+      url: 'https://www.figma.com/design/s9VKYHFn1GncFyxd5l19nU/1.11-Amsterdam?node-id=6086-1559&node-type=canvas&m=dev',
     },
   },
   argTypes: {
-    datePickerType: {
-      options: createSelectOptions(DATE_PICKER_TYPES),
-      control: { type: 'select', labels: { null: DATE_PICKER_TYPES.SINGLE } },
-      table: {
-        defaultValue: { summary: DATE_PICKER_TYPES.SINGLE },
-      },
-    },
-    size: {
-      options: ['sm', 'md', 'lg'],
+    locale: { control: { type: 'text' } },
+    dateFormat: {
+      options: [
+        'Y-m-d',
+        'm-d-Y',
+        'd-m-Y',
+        'Y-m-d H:i',
+        'Y-m-d H:i:s',
+        'm-d-Y H:i:s',
+        'd-m-Y H:i:s',
+      ],
       control: { type: 'select' },
     },
-    minDate: {
-      control: { type: 'text' },
-    },
-    maxDate: {
-      control: { type: 'text' },
-    },
-    step: {
-      control: { type: 'text' },
+    mode: {
+      options: ['single', 'multiple'],
+      control: { type: 'select' },
     },
+    minDate: { control: { type: 'text' } },
+    maxDate: { control: { type: 'text' } },
   },
 };
 
-const args = {
-  unnamed: 'Date',
-  size: 'md',
-  name: 'datepicker',
+const disconnectFlatpickr = () => {
+  const calendarElements = document.querySelectorAll('.flatpickr-calendar');
+  calendarElements.forEach((calendar) => calendar.remove());
+};
+
+const InputTemplate = (args) => {
+  useEffect(() => {
+    return () => {
+      disconnectFlatpickr();
+    };
+  }, []);
+
+  return html`<kyn-date-picker
+    .nameAttr="${args.nameAttr}"
+    .locale="${args.locale}"
+    .dateFormat="${args.dateFormat}"
+    ?required="${args.required}"
```

Comment:
If you update the `patterns/forms` page with these components set as required, it will be easier to test the validation with form submit.

Reply:
✅